drivers/ethernet/eth_stellaris.c

```diff
@@ -0,0 +1,365 @@
+		SYS_LOG_ERR("No data to send");
+		net_pkt_unref(pkt);
+		return -ENODATA;
+	}
+
+	/* Frame transmission
+	 *  First two bytes are data_len for frame,
+	 *  Initially send the data_len
+	 */
+	data_len = net_pkt_get_len(pkt);
+	eth_stellaris_send_byte(dev, data_len & 0xff);
+	eth_stellaris_send_byte(dev, (data_len & 0xff00) >> 8);
+
+	/* Send the  header, header is 14 bytes */
+	head_len_left = net_pkt_ll_reserve(pkt);
+	eth_hdr = net_pkt_ll(pkt);
+	for (i = 0; i < head_len_left; ++i) {
+		eth_stellaris_send_byte(dev, eth_hdr[i]);
+	}
+
+	/* Send the payload */
+	for (frag = pkt->frags; frag; frag = frag->frags) {
+		for (i = 0; i < frag->len; ++i) {
+			eth_stellaris_send_byte(dev, frag->data[i]);
+		}
+	}
+	/* will transmit the partial word */
+	eth_stellaris_flush(dev);
+
+	/* Enable Txn */
+	sys_write32(BIT_MACTR_NEWTX, REG_MACTR);
+
+	/* Wait and check if txn successful or not */
+	k_sem_take(&dev_data->tx_sem, K_FOREVER);
+
+	if (dev_data->tx_err) {
+		SYS_LOG_ERR("Txn failed");
+		dev_data->tx_err = false;
+		net_pkt_unref(pkt);
+		return -EIO;
+	}
+
+	/* Update statistics counters */
+	eth_stats_update_bytes_tx(iface, net_pkt_get_len(pkt));
+	eth_stats_update_pkts_tx(iface);
+	pkt_hdr = NET_ETH_HDR(pkt);
+	if (net_eth_is_addr_multicast(&pkt_hdr->dst)) {
+		eth_stats_update_multicast_tx(iface);
+	} else if (net_eth_is_addr_broadcast(&pkt_hdr->dst)) {
+		eth_stats_update_broadcast_tx(iface);
+	}
+
+	SYS_LOG_DBG("pkt send %p len %d", pkt, net_pkt_get_len(pkt));
+	net_pkt_unref(pkt);
+
+	return 0;
+}
+
+void eth_stellaris_rx_error(struct net_if *iface)
+{
+	u32_t val;
+
+	struct device *dev = net_if_get_device(iface);
+
+	eth_stats_update_errors_rx(iface);
+	/* Clear the rx_frame buffer,
+	 * otherwise it could lead to underflow errors
+	 */
+	sys_write32(0x0, REG_MACRCTL);
+	sys_write32(BIT_MACRCTL_RSTFIFO, REG_MACRCTL);
+	val = BIT_MACRCTL_BADCRC | BIT_MACRCTL_RXEN;
+	sys_write32(val, REG_MACRCTL);
+}
+
+static int eth_stellaris_rx_pkt(struct device *dev, struct net_pkt *pkt)
+{
+	u32_t reg_val;
+	int frame_len, bytes_left;
+
+	/*Read the First Word */
+	reg_val = sys_read32(REG_MACDATA);
+	frame_len = reg_val & 0x0000ffff;
+	if (!net_pkt_append(pkt, 2, (u8_t *)&reg_val + 2, K_NO_WAIT)) {
+		return -1;
+	}
+
+	/* A word have been read already, Thus minus 4 bytes to be read */
+	bytes_left = frame_len - 4;
+
+	/* Read the rest of words, minus the partial word and FCS byte */
+	for (; bytes_left > 7; bytes_left -= 4) {
+		reg_val = sys_read32(REG_MACDATA);
+		if (!net_pkt_append(pkt, 4, (u8_t *)&reg_val, K_NO_WAIT)) {
+			return -1;
+		}
+	}
+
+	/* Handle the last partial word and Discard the 4 Byte FCS */
+	while (bytes_left > 0) {
+		reg_val = sys_read32(REG_MACDATA);
+		/* Discard the last FCS word */
+		if (bytes_left <= 4) {
+			bytes_left = 0;
+			break;
+		}
+		/* Read the the partial word */
+		if (!net_pkt_append(pkt, bytes_left - 4,
+				    (u8_t *)&reg_val, K_NO_WAIT)) {
+			return -1;
+		}
+		bytes_left -= 4;
```

Comment:
Place empty lines after } as that is the style in zephyr.

Reply:
Done.